zvariant/src/lib.rs

```diff
@@ -1293,6 +1293,354 @@ mod tests {
+    #[test]
+    fn dict_iter_str_signature() {
+        let mut dict = Dict::new(<&str>::signature(), <Signature<'_>>::signature());
+        let values = vec![
+            ("123", Signature::try_from("i").unwrap()),
+            ("456", Signature::try_from("u").unwrap()),
+        ];
+        for (k, v) in values.iter() {
+            dict.add(k, v).unwrap();
+        }
+        let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+        for (k, v) in values.iter() {
+            expect.push((Value::from(k), Value::from(v)));
+        }
+        let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+        let actual = dict.iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&mut dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = dict.into_iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect);
+    }
+
+    #[cfg(unix)]
+    #[test]
+    fn dict_iter_fd_str() {
+        let mut dict = Dict::new(<Fd<'_>>::signature(), <&str>::signature());
+        for (k, v) in vec![(mock_fd(0), "123"), (mock_fd(1), "456")] {
+            dict.add(k, v).unwrap();
+        }
+        let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+        for (k, v) in vec![(mock_fd(0), "123"), (mock_fd(1), "456")] {
+            expect.push((Value::from(k), Value::from(v)));
+        }
+        let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+        let actual = dict.iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&mut dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = dict.into_iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect);
+    }
+
+    #[cfg(unix)]
+    #[test]
+    fn dict_iter_str_fd() {
+        let mut dict = Dict::new(<&str>::signature(), <Fd<'_>>::signature());
+        for (k, v) in vec![("123", mock_fd(0)), ("456", mock_fd(1))] {
+            dict.add(k, v).unwrap();
+        }
+        let mut expect = Vec::<(Value<'_>, Value<'_>)>::new();
+        for (k, v) in vec![("123", mock_fd(0)), ("456", mock_fd(1))] {
+            expect.push((Value::from(k), Value::from(v)));
+        }
+        let expect_iter = expect.iter().map(|(k, v)| (k, v)).collect::<Vec<_>>();
+        let actual = dict.iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = (&mut dict).iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect_iter);
+        let actual = dict.into_iter().collect::<Vec<_>>();
+        assert_eq!(actual, expect);
+    }
+
+    #[test]
+    fn dict_iter_mut_u8_str() {
```

Comment:
we especially don't want separate tests just for `mut`.

Reply:
Moved to `dict_value()` test case.